Sources/swift-bundler/Bundler/VisionOSBundler.swift

```diff
@@ -0,0 +1,260 @@
+
+    // let createAppIconIfPresent: () -> Result<Void, VisionOSBundlerError> = {
+    //   if let path = appConfiguration.icon {
+    //     let icon = packageDirectory.appendingPathComponent(path)
+    //     return Self.createAppIcon(icon: icon, outputDirectory: appAssets)
+    //   }
+    //   return .success()
+    // }
+
+    let copyResourcesBundles: () -> Result<Void, VisionOSBundlerError> = {
+      ResourceBundler.copyResources(
+        from: productsDirectory,
+        to: appBundle,
+        fixBundles: !isXcodeBuild && !universal,
+        platform: targetingSimulator ? .visionOSSimulator : .visionOS,
+        platformVersion: platformVersion,
+        packageName: packageName,
+        productName: appConfiguration.product
+      ).mapError { error in
+        .failedToCopyResourceBundles(error)
+      }
+    }
+
+    let copyDynamicLibraries: () -> Result<Void, VisionOSBundlerError> = {
+      DynamicLibraryBundler.copyDynamicLibraries(
+        dependedOnBy: appExecutable,
+        to: appDynamicLibrariesDirectory,
+        productsDirectory: productsDirectory,
+        isXcodeBuild: false,
+        universal: false,
+        makeStandAlone: false
+      ).mapError { error in
+        .failedToCopyDynamicLibraries(error)
+      }
+    }
+
+    let embedProfile: () -> Result<Void, VisionOSBundlerError> = {
+      if let provisioningProfile = provisioningProfile {
+        return embedProvisioningProfile(provisioningProfile, in: appBundle)
+      } else {
+        return .success()
+      }
+    }
+
+    let codesign: () -> Result<Void, VisionOSBundlerError> = {
+      if let identity = codesigningIdentity {
+        return CodeSigner.signWithGeneratedEntitlements(
+          bundle: appBundle,
+          identityId: identity,
+          bundleIdentifier: appConfiguration.identifier
+        ).mapError { error in
+          .failedToCodesign(error)
+        }
+      } else {
+        return .success()
+      }
+    }
+
+    // TODO: Why is app icon creation disabled?
+    let bundleApp = flatten(
+      { createAppDirectoryStructure(at: outputDirectory, appName: appName) },
+      { copyExecutable(at: executableArtifact, to: appExecutable) },
+      {
+        createMetadataFiles(
+          at: appBundle,
+          appName: appName,
+          appConfiguration: appConfiguration,
+          visionOSVersion: platformVersion,
+          targetingSimulator: targetingSimulator
+        )
+      },
+      // { createAppIconIfPresent() },
+      { copyResourcesBundles() },
+      { copyDynamicLibraries() },
+      { embedProfile() },
+      { codesign() }
+    )
+
+    return bundleApp().mapError { (error: VisionOSBundlerError) -> Error in
+      error
+    }
+  }
+
+  // MARK: Private methods
+
+  /// Creates the directory structure for an app.
+  ///
+  /// Creates the following structure:
+  ///
+  /// - `AppName.app`
+  ///   - `Contents`
+  ///     - `MacOS`
+  ///     - `Resources`
+  ///     - `Libraries`
```

Comment:
Update this for the visionOS structure:
- AppName.app
  - Libraries
  
(I realise that this was just copied from the IOSBundler so that probably needs updating too)